meetings/2021-11-23.md

```diff
@@ -0,0 +1,115 @@
+* ...: You have Data Registry and Data Registration for medical records. Medical record is composed of prescriptions and appointments and conditions. Medical record is more of a meta object which just links to all those other things.
+* ...: If I want to share the whole medical record with someone who helps taking care of me, I can share the whole medical record with them. To give partial access, in that case Alice wants to use Prescripto application. She wants it to access prescriptions but not conditions, appointments etc. Prescripto needs to see all the prescriptions and also if it adds new one it needs to link it to medical record. For prescripto to know it needs to be able to see those links. If Medical Record contains links to everything (conditions, appointments, prescriptions). Even that Prescripto can't follow those links, it still leaks data by knowing how many conditions and appointments there are linked. Which on itself is not acceptable.
+* ...: We either need to say that Prescrito gets partial access to medical record, which would require something like shapemasks or tripple based authorization. Or we need to maintain those links in stand alone resources. This term of reference lists comes from that. We could say that Prescripto has access to prescriptions reference list but not to reference lists of conditions and appointments. It seems like the most straight forward approach. We would need to ensure that shape paths can follow external resources.
+
+* Pavlik: Ties into what's the parent shape tree and the target shape tree (e.g. child). Needs to be a way in the medical record to say what the reference lists are for the children. With that information we'd be able to discover what the right links are.
+* Justin: I think the first place to investigate would be in shape tree definition. Shape trees are designed to provide that navigation of multi resource hierarchy. We already have the ground work for that. I can see where this notion of reference lists typo of a model could fit in.
+* Eric: I think we could do it with shape masks.
+* Justin: I think shape masks are elegant long term.
+* Eric: When you want to PUT you need to know which part of the graph.
+* Pavlik: Might be useful to talk to GraphMetrix about triple level access control approach. Would be good to get some feedback from their implementation.
+* Justin: I believe that shape masks are so complementary to the rest of the stuff you need to do as application. It may be because we use shape trees in interop.
+* ...: I plan on expanding this use case a little bit more. I could try to design a shape tree that tries to map this.
+* Pavlik: What's important is to keep the reference list as a detail of the reference itself and not make some separate reference to it.
+* Justin: So - Encode the external reference (to the list) into the existing shape tree reference between medical record and prescription (Pavlik: ACK)
```

Comment:
```suggestion
* Eric: When you want to `PUT`, you need to know which part of the graph.
* Pavlik: Might be useful to talk to GraphMetrix about triple-level access control approach. Would be good to get some feedback from their implementation.
* Justin: I believe that shape masks are complementary to the rest of the stuff you need to do as application. It may be because we use shape trees in interop.
  * I plan on expanding this use case a little bit more. I could try to design a shape tree that tries to map this.
* Pavlik: What's important is to keep the reference list as a detail of the reference itself and not make some separate reference to it.
* Justin: So — Encode the external reference (to the list) into the existing shape tree reference between medical record and prescription? (Pavlik: ACK)
```